src/main/java/io/gravitee/policy/jwt/JWTPolicy.java

```diff
@@ -142,25 +158,21 @@ private void setAuthContextInfos(HttpExecutionContext ctx, LazyJWT jwt, JWTClaim
         if (configuration.isExtractClaims()) {
             ctx.setAttribute(CONTEXT_ATTRIBUTE_JWT_CLAIMS, claims.getClaims());
         }
-
-        if (!configuration.isPropagateAuthHeader()) {
-            request.headers().remove(HttpHeaders.AUTHORIZATION);
-        }
     }
 
-    private Maybe<LazyJWT> extractToken(HttpExecutionContext ctx) {
-        Optional<String> token = TokenExtractor.extract(ctx.request());
+    private Single<LazyJWT> extractToken(BaseExecutionContext ctx) {
```

Comment:
To avoid confusion between `extractSecurityToken` and `extractToken`, I would rename it to `fetchJWTToken`.

That way we better understand the purpose and naming is less confusing:
- `extractSecurityToken`: results in returning a SecurityToken (if any)
- `fetchJWTToken`: results in getting the JWT token from the context.